sos/report/plugins/juju.py

```diff
@@ -53,6 +65,52 @@ def setup(self):
             # logs in the directory.
             self.add_copy_spec("/var/log/juju/*.log")
 
+        # Only run the feature code if this plugin option is set
+        if not self.get_option("is-feature"):
+            return
+
+        juju_user = self.get_option("juju-user")
+        try:
+            pwd.getpwnam(juju_user)
+        except KeyError:
+            # The user doesn't exist, this will skip the rest
+            self._log_warn(f'User "{juju_user}" does not exist, will not collect Juju information.')
+            return
+
+        user_context = f'sudo -u {juju_user}'
```

Comment:
Use the `runas` option for `add_cmd_output()`, `collect_cmd_output()`, and `exec_cmd()` to run as non-root. We don't embed `sudo` into command collections.